### DISCUSSION TITLE
Why does an RA-TLS certificate contain a certificate chain?

### DISCUSSION THREAD

**Discussion by fwoodruff** [gramineproject/gramine#1650] [Q&A]
I was having a look at the guts of an RA-TLS certificate. I notice it contains a TLS certificate chain. Is that certificate chain used? What does it add security-wise vs just checking the SGX quote? It dramatically increases the size of RA-TLS certificates.

**Comment by dimakuv** [comment]
> I notice it contains a TLS certificate chain

Where exactly? There are two places:
1. The X.509 certificate's chain (starting from the `ISSUER` field). But this chain is degenerative -- it has a single entry, which is this cert itself. That's why we call the RA-TLS certificate self-signed (the chain ends in itself).
2. The SGX Quote's chain. See the picture here: https://gramine.readthedocs.io/en/stable/attestation.html#mid-level-ra-tls-interface. Note that this chain is **not** a proper X.509 cert's chain -- it's an ad-hoc format of the SGX Quote. This SGX Quote's chain of certificates is required to trace the SGX Quote to the Intel hardware -- the SGX Quote is generated by a special Intel-distributed Quoting Enclave (which has its own certificate), which in turn is bootstrapped by a special Intel-distributed Provisioning Certification Enclave (which has its own certificate of one level higher), which in turn is bootstrapped by communicating via internet to the Intel Provisioning Certification Service (which has its own certificate of again one level higher). So you have a chain of 3 certificates (there may be 4 certificates in the chain, this detail I don't remember).
